test/basicfuns.jl

```diff
@@ -88,15 +88,26 @@ end
                  ([-Inf, Inf], Inf),
                  ([-Inf, 9.0], 9.0),
                  ([Inf, 9.0], Inf),
-                 ([NaN, 9.0], NaN),  # NaN propagation
-                 ([NaN, Inf], NaN),  # NaN propagation
-                 ([NaN, -Inf], NaN), # NaN propagation
                  ([0, 0], log(2.0))] # non-float arguments
         for (arguments, result) in cases
             @test logaddexp(arguments...) ≡ result
             @test logsumexp(arguments) ≡ result
         end
     end
+    let nan_cases = [[NaN, 9.0], [NaN, Inf], [NaN, -Inf]] # NaN propagation
+        for args in nan_cases
+            @test isnan(logaddexp(args...))
+            @test isnan(logsubexp(args...))
+            @test isnan(logsumexp(args))
+        end
+    end
```

Comment:
This isn't that many cases, so it might be easier just to write them all out.

Reply:
Ok, done. I was just trying to be consistent with the pattern of the tests above, but I agree with you.